gapis/perfetto/android/trace.go

```diff
@@ -356,7 +380,26 @@ func (p *Process) captureWithClientApi(ctx context.Context, start task.Signal, s
 	}
 
 	var numWritten int64
-	if downloadWhileTracing {
+	if p.mode == traceIntoFile {
+		tmp, err := file.Temp()
+		if err != nil {
+			return 0, log.Err(ctx, err, "Failed to create a temp file")
+		}
+		if err := p.device.Pull(ctx, perfettoTraceFile, tmp.System()); err != nil {
+			return 0, err
```

Comment:
nit: just wondering whether it's intended here to drop out error message extra wrapping.

Reply:
If the pull fails, it's error message is sensible. On the other hand, the file system errors would be kind of surprising, so they are wrapped.